src/ui/pause_menu.rs

```diff
@@ -275,3 +271,16 @@ fn main_pause_menu(
         }
     });
 }
+
+/// Helper for pausing session depending on if online or offline match.
+/// `remain_inactive` is used if other systems like scoring have set session inactive and wish to
+/// prevent that change.
+fn pause_session(paused: bool, is_online: bool, session: &mut Session, remain_inactive: bool) {
+    if is_online {
+        // Online session remains active so that it will not timeout with other players.
+        // Instead we disable the input so it is not captured by game while paused.
+        session.runner.disable_local_input(paused);
+    } else if !paused && !remain_inactive {
```

Comment:
whoops looks like a bug here, need to make sure branch for un-pausing local sessions is present.